components/post_priority/post_priority_picker_overlay.tsx

```diff
@@ -5,17 +5,17 @@ import React, {memo} from 'react';
 import {Overlay} from 'react-bootstrap';
 import memoize from 'memoize-one';
 
-import {PostPriority} from '@mattermost/types/posts';
-
 import {popOverOverlayPosition} from 'utils/position_utils';
 
+import {Post} from '@mattermost/types/posts';
+
 import PostPriorityPicker from './post_priority_picker';
 
 type Props = {
     show: boolean;
-    priority?: PostPriority;
+    settings?: Post['metadata']['priority'];
```

Comment:
I think we could consider using `PostPriorityMetadata` directly in places like these since that's a bit simpler to type and read